protostar/cheatable_starknet/callable_hint_locals/declare_hint_local.py

```diff
@@ -84,11 +92,37 @@ def _make_contract_class(contract_identifier: ContractIdentifier):
 
             return ContractClass.load(sierra_compiled)
 
-        def _make_compiled_class(contract_identifier: ContractIdentifier):
-            casm_compiled = self._project_compiler.compile_contract_to_casm_from_contract_identifier(
-                contract_identifier
+        def _make_compiled_class(contract_name: str, contract_path: Path):
+            casm_compiled = self._compile_to_casm(
+                contract_name=contract_name, contract_path=contract_path
             )
 
             return CompiledClass.loads(casm_compiled)
 
         return declare
+
+    def _compile_to_sierra(self, contract_name: str, contract_path: Path):
+        try:
+            sierra_compiled = compile_starknet_contract_to_sierra_from_path(
+                contract_path
+            )
+
+            if sierra_compiled is None:
+                raise TypeError
+        except (CairoBindingException, TypeError) as ex:
+            raise CheatcodeException(
+                self, f"Compilation of contract {contract_name} to sierra failed"
+            ) from ex
+        return sierra_compiled
+
+    def _compile_to_casm(self, contract_name: str, contract_path: Path):
+        try:
+            casm_compiled = compile_starknet_contract_to_casm_from_path(contract_path)
+
+            if casm_compiled is None:
+                raise TypeError
+        except (CairoBindingException, TypeError) as ex:
+            raise CheatcodeException(
+                self, f"Compilation of contract {contract_name} to casm failed"
+            ) from ex
+        return casm_compiled
```

Comment:
Kind of unsure about this: We have to both check if calling bindings doesn't raise an exception and check if value returned is not None, and this seemed like the most concise way of doing that. But the code is quite ugly.

Reply:
I think we always want to raise an error, not return `None` (correct me if i'm wrong). Perhaps we should modify how the binding works?

Reply:
Related cairo compiler PR https://github.com/software-mansion-labs/cairo/pull/71